lib/Settings/Admin.php

```diff
@@ -21,25 +21,31 @@
 
 namespace OCA\LogReader\Settings;
 
+use OCA\LogReader\Service\SettingsService;
 use OCP\AppFramework\Http\TemplateResponse;
-use OCP\IL10N;
+use OCP\AppFramework\Services\IInitialState;
 use OCP\Settings\IDelegatedSettings;
+use OCP\Util;
 
 class Admin implements IDelegatedSettings {
-	/** @var IL10N */
-	private $l;
+	private IInitialState $initialState;
+	private SettingsService $settingsService;
 
-	public function __construct(IL10N $l) {
-		$this->l = $l;
+	public function __construct(IInitialState $initialState, SettingsService $settingsService) {
+		$this->initialState = $initialState;
+		$this->settingsService = $settingsService;
 	}
+
 	/**
 	 * @return TemplateResponse
 	 */
 	public function getForm() {
-		return new TemplateResponse('logreader',
-			'index',
-			['appId' => 'logreader', 'inline-settings' => 'true'],
-			'');
+		Util::addScript('logreader', 'logreader-main');
```

Comment:
Same thing with $this->appName if possible

Reply:
On settings there is no `appName` should I change the constructor to use DI to inject it and use it afterwards?

Reply:
No I thought this was a `Controller` as well. If you do not already have a var for it do as you wish.